notebooks/Numba_054_release_demo.ipynb

```diff
@@ -0,0 +1,240 @@
+    "* ``np.isscalar`` ([@guilhermeleobas](https://github.com/guilhermeleobas))\n",
+    "* ``np.isneginf`` ([@guilhermeleobas](https://github.com/guilhermeleobas))\n",
+    "* ``np.isposinf`` ([@guilhermeleobas](https://github.com/guilhermeleobas))\n",
+    "\n",
+    "and also enhances the support for:\n",
+    "\n",
+    "* ``np.argmax()`` now supports the `axis` kwarg ([@itamarst](https://github.com/itamarst))\n",
+    "* ``__setitem__`` using ``0d`` arrays ([@guilhermeleobas](https://github.com/guilhermeleobas))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from numba import literal_unroll\n",
+    "\n",
+    "@njit\n",
+    "def demo_numpy_new_features():\n",
+    "    x = np.linspace(0, 10, 20)\n",
+    "    print(\"np.clip\\n\", np.clip(x, 2., 5.))\n",
+    "    print(\"np.rot90\\n\", np.rot90(np.arange(4).reshape((2, 2))))\n",
+    "    print(\"np.swapaxes\\n\", np.swapaxes(np.arange(8).reshape((2, 2, 2)), 0, 2))\n",
+    "\n",
+    "    np.random.seed(0)\n",
+    "    print(\"np.random.dirichlet\\n\", np.random.dirichlet((2, 3, 4), 2))\n",
+    "\n",
+    "    for x in literal_unroll((1, 1j, np.array([1, 2j, 3]))):\n",
+    "        print(\"x is \", x)\n",
+    "        print(\"  np.iscomplexobj(x)\", np.iscomplexobj(x))\n",
+    "        print(\"  np.isreal(x)\", np.isreal(x))\n",
+    "        print(\"  np.isrealobj(x)\", np.isrealobj(x))\n",
+    "        print(\"  np.isscalar(x)\", np.isscalar(x))\n",
+    "    for x in literal_unroll((1, np.inf, -np.inf, np.nan)):\n",
+    "        print(\"x is \", x)\n",
+    "        print(\"  np.isneginf(x)\", np.isneginf(x))\n",
+    "        print(\"  np.isposinf(x)\", np.isposinf(x))\n",
+    "        print(\"  np.iscomplex(x)\", np.iscomplex(x))\n",
+    "        \n",
+    "demo_numpy_new_features()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Enhancements:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@njit\n",
+    "def demo_numpy_enhancements():\n",
+    "    # Axis support in np.argmax\n",
+    "    x = np.arange(12).reshape((3, 2, 2))\n",
+    "    x[:, 1, :] = 100\n",
+    "    print(\"np.argmax(x, axis=2)\\n\", np.argmax(x, axis=2))\n",
+    "    \n",
+    "    # Setitem using 0d arrays\n",
+    "    idx = np.array(1)\n",
+    "    x = np.arange(10)\n",
+    "    x[idx] = 100\n",
+    "    print(\"Setitem of x[idx]\\n\", x, \", idx =\", idx, \", idx.ndim =\", idx.ndim)\n",
+    "    \n",
+    "demo_numpy_enhancements()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "----\n",
+    "\n",
+    "## See Also\n",
+    "\n",
+    "- [0.54 CUDA Demo](https://mybinder.org/v2/gh/numba/numba-examples/master?filepath=notebooks%2FNumba_054_CUDA_Release_Demo.ipynb)\n",
+    "- [0.54 TODO involved features demo, e.g. array subclass](TODO)"
```

Comment:
This needs doing!

Reply:
It's at https://github.com/numba/numba-examples/pull/38